src/System.Collections.Concurrent/tests/BlockingCollectionCancellationTests.cs

```diff
@@ -21,17 +22,18 @@ public static void InternalCancellation_CompleteAdding_Negative()
             //call Take.. it should wake up with an OCE. when CompleteAdding() is called.
             Assert.Throws<InvalidOperationException>(
                () => coll1.Take());
-               // "InternalCancellation_WakingUpTake:  an IOE should be thrown if CompleteAdding occurs during blocking Take()");
+            // "InternalCancellation_WakingUpTake:  an IOE should be thrown if CompleteAdding occurs during blocking Take()");
             Assert.Throws<InvalidOperationException>(
                 () => coll1.Add(1));
-                // "InternalCancellation_WakingUpAdd:  an InvalidOpEx should be thrown if CompleteAdding occurs during blocking Add()");
+            // "InternalCancellation_WakingUpAdd:  an InvalidOpEx should be thrown if CompleteAdding occurs during blocking Add()");
             Assert.Throws<InvalidOperationException>(
                () => coll1.TryAdd(1, 1000000));  //an indefinite wait to add.. 1000 seconds.
-               // "InternalCancellation_WakingUpTryAdd:  an InvalidOpEx should be thrown if CompleteAdding occurs during blocking Add()");
+            // "InternalCancellation_WakingUpTryAdd:  an InvalidOpEx should be thrown if CompleteAdding occurs during blocking Add()");
         }
 
         //This tests that Take/TryTake wake up correctly if CompleteAdding() is called while the taker is waiting.
         [Fact]
+        [OuterLoop]
```

Comment:
I marked a test [OuterLoop] not only based on the run time, but the scenarios. For example, I've marked the all negative tests as OuterLoop. I think we should only run basic scenarios as inner loop tests. When we hook up the code coverage run, I think we can use that data to guide us about what and when run tests.